tests/tests_pytorch/test_cli.py

```diff
@@ -290,31 +314,31 @@ def test_lightning_cli_save_config_only_once(tmpdir):
     cli.trainer.test(cli.model)  # Should not fail because config already saved
 
 
-def test_lightning_cli_config_and_subclass_mode(tmpdir):
+def test_lightning_cli_config_and_subclass_mode(cleandir):
     input_config = {
         "fit": {
             "model": {"class_path": "pytorch_lightning.demos.boring_classes.BoringModel"},
             "data": {
-                "class_path": "pytorch_lightning.demos.boring_classes.BoringDataModule",
-                "init_args": {"data_dir": str(tmpdir)},
+                "class_path": "DataDirDataModule",
+                "init_args": {"data_dir": "."},
             },
-            "trainer": {"default_root_dir": str(tmpdir), "max_epochs": 1, "enable_model_summary": False},
+            "trainer": {"max_epochs": 1, "enable_model_summary": False},
         }
     }
-    config_path = tmpdir / "config.yaml"
+    config_path = "config.yaml"
     with open(config_path, "w") as f:
         f.write(yaml.dump(input_config))
 
-    with mock.patch("sys.argv", ["any.py", "--config", str(config_path)]):
+    with mock.patch("sys.argv", ["any.py", "--config", config_path]):
         cli = LightningCLI(
             BoringModel,
             BoringDataModule,
             subclass_mode_model=True,
             subclass_mode_data=True,
-            trainer_defaults={"callbacks": LearningRateMonitor()},
+            save_config_kwargs={"overwrite": True},
```

Comment:
I noticed one small annoyance. If a logger is not set (which would write configs to a new `version_x` directory), then passing a config forces the user to overwrite it even if the contents are the same.

Maybe we should avoid the error if they would be equal. cc @mauvilsa

Reply:
I am not sure what you mean. If no logger is set by the user, then in the saved config the value will be `null`. Why is it that the user is forced to overwrite?